### PR TITLE
Ft:lougout 184638376

### PR DESCRIPTION
**Pull request by dianepretty**
#### What does this PR do?
It adds the logout feature
 ...

#### Description of  the task to be completed?
The user will be able to logout
...

#### How should this be manually tested?
get all tokens "/api/users/tokens"
update toke "/api/users/logout/id" where id is the id of the token to inactivate
...

#### Any background context you want to provide?

...

#### What are the relevant pivotal tracker stories?
##184638376
...

#### Screenshots (if appropriate)
image
image

...

### CODE REVIEW

File: src/controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      return res.status(400).json({ message: 'Invalid credentials' });
+    }
+    const isMatch = await user.isValidPassword(password);
+    if (!isMatch) {
+      return res.status(400).json({ message: 'Invalid credentials' });
+    }
+    const token = jwt.sign({ id: user.id }, process.env.JWT_SECRET, {
+      expiresIn: process.env.JWT_EXPIRES_IN
+    });
+    await Token.create({ userId: user.id, state: 'active' });
```
Comment (oreste-abizera) [inline src/controllers/userController.js:71]: You should store that token in the database as well

File: src/controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    return res.status(400).json({ tokens });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ message: 'Server Error', error: error.message });
+  }
+}
+export async function logout (req, res) {
+  const id = req.params.id;
+  try {
+    const token = await Token.findOne({ where: { id } });
```
Comment (oreste-abizera) [inline src/controllers/userController.js:93]: You need to find by auth token passed in the headers, not by the user id

**Review by oreste-abizera — CHANGES_REQUESTED** [review]
Great work on the implemented work! However, we must also ensure that the generated token is added to the database as well. Additionally, in order to enable logging out functionality, we'll need to create a middleware that checks whether the user is authorized. Also in the logout function, we can use the auth token passed in headers to find the associated token instead of using the user id.

**Review by honorine22 — APPROVED** [review]

**Review by oreste-abizera — APPROVED** [review]

**Review by felixdusengimana — CHANGES_REQUESTED** [review]
Can you add swagger documentation of your API

**Review by Nkbtemmy — APPROVED** [review]

**Pull request merged**